src/components/layout/MainLayout/MainLayout.js

```diff
@@ -5,17 +5,41 @@ import Header from '../Header/Header';
 import Footer from '../Footer/Footer';
 import ProductsCompare from '../ProductsCompare/ProductsCompareContainer';
 
-const MainLayout = ({ children }) => (
-  <div>
-    <Header />
-    {children}
-    <Footer />
-    <ProductsCompare />
-  </div>
-);
+class MainLayout extends React.Component {
+  checkMode = val => {
+    return val <= 767 ? 'mobile' : val <= 1024 ? 'tablet' : 'desktop';
+  };
+
+  componentWillMount() {
+    const { setRwdMode } = this.props;
+    window.addEventListener('resize', e => {
+      const mode = this.checkMode(e.target.window.innerWidth);
+      setRwdMode(mode);
```

Comment:
Zbyt często się to wywołuje, akcja powinna być wywoływana tylko jeśli zmienia się tryb, jeśli się nie zmienia to po co dispatchować akcję :)

Reply:
Dodałem zmienną currentRwdMode i porównóje do niej dopiero jak nastepuje zmiana nadpisuje currentRwdMode i dispatchuje. Wygląda na to że działa.